File: 2816-double-a-number-represented-as-a-linked-list/2816-double-a-number-represented-as-a-linked-list.java
class Solution {
    public ListNode doubleIt(ListNode head) {
        var carry = doubleItt(head);
        if(carry > 0) head = new ListNode(carry, head);
        return head;
    }

    private int doubleItt(ListNode node){
        if(node == null) return 0;
        var carry = doubleItt(node.next);
        var num = carry + node.val * 2;
        node.val = num%10;
        carry = num / 10;
        return carry;
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

File: 2816-double-a-number-represented-as-a-linked-list/2816-double-a-number-represented-as-a-linked-list.cs
public class Solution {
    public ListNode DoubleIt(ListNode head) {
        var carry = DoubleItt(head);
        if(carry > 0)
            head = new ListNode(carry, head);
        return head;
    }

    private int DoubleItt(ListNode node){
        if(node == null) return 0;
        var carry = DoubleItt(node.next);
        var num = carry + node.val * 2;
        node.val = num%10;
        carry = num / 10;
        return carry;
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */